src/main/java/org/codeforamerica/messaging/repositories/MessageRepository.java

```diff
@@ -13,18 +13,16 @@ public interface MessageRepository extends CrudRepository<Message, Long> {
 
     @Query("""
             SELECT
-             SUM(CASE email.status WHEN 'accepted' THEN 1 ELSE 0 END) AS acceptedEmailCount,
-             SUM(CASE email.status WHEN 'rejected' THEN 1 ELSE 0 END) AS rejectedEmailCount,
-             SUM(CASE email.status WHEN 'delivered' THEN 1 ELSE 0 END) AS deliveredEmailCount,
-             SUM(CASE email.status WHEN 'failed' THEN 1 ELSE 0 END) AS undeliveredEmailCount,
-             SUM(CASE sms.status WHEN 'accepted' THEN 1 ELSE 0 END) AS acceptedSmsCount,
-             SUM(CASE sms.status WHEN 'rejected' THEN 1 ELSE 0 END) AS rejectedSmsCount,
-             SUM(CASE sms.status WHEN 'delivered' THEN 1 ELSE 0 END) AS deliveredSmsCount,
-             SUM(CASE sms.status WHEN 'undelivered' THEN 1 ELSE 0 END) AS undeliveredSmsCount
+             SUM(CASE emailStatus WHEN 'accepted' THEN 1 ELSE 0 END) AS acceptedEmailCount,
+             SUM(CASE emailStatus WHEN 'rejected' THEN 1 ELSE 0 END) AS rejectedEmailCount,
+             SUM(CASE emailStatus WHEN 'delivered' THEN 1 ELSE 0 END) AS deliveredEmailCount,
+             SUM(CASE emailStatus WHEN 'failed' THEN 1 ELSE 0 END) AS undeliveredEmailCount,
+             SUM(CASE smsStatus WHEN 'accepted' THEN 1 ELSE 0 END) AS acceptedSmsCount,
+             SUM(CASE smsStatus WHEN 'rejected' THEN 1 ELSE 0 END) AS rejectedSmsCount,
+             SUM(CASE smsStatus WHEN 'delivered' THEN 1 ELSE 0 END) AS deliveredSmsCount,
+             SUM(CASE smsStatus WHEN 'undelivered' THEN 1 ELSE 0 END) AS undeliveredSmsCount
```

Comment:
Just spotted this inconsistency. Slight preference for `failed` in both over `undelivered`.
> 'failed' AS undeliveredEmailCount
> 'undelivered' AS undeliveredSmsCount

Reply:
Great catch. The issue is that `failed` is returned by `Mailgun` for undelivered emails and this query deals with that anomaly. However, I introduced `failed` in this story for provider handoff failures without realizing that it was already used. I should replace the new `failed` with `provider_rejected` or similar :-(